core/src/world/place/building/religious/shrine.rs

```diff
@@ -0,0 +1,192 @@
+fn deity(rng: &mut impl Rng) -> String {
+    match rng.gen_range(0..10) {
+        0..=1 => format!("The {}", person(rng)),
+        2 => format!("The {} {}", descriptor(rng), person(rng)),
+        3..=4 => format!("The {}", animal(rng)),
+        5 => format!("The {} {}", descriptor(rng), animal(rng)),
+        6..=8 => concept(rng).to_string(),
+        9 => format!("{} {}", descriptor(rng), concept(rng)),
+        _ => unreachable!(),
+    }
+}
+//PERSON
+fn person(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const PEOPLE: &[&str] = &[
+        "Father","Mother","Parent","Sibling","Hunter","Emperor","Empress","Warrior","Sage","Ancestor"
+    ];
+    PEOPLE[rng.gen_range(0..PEOPLE.len())]
+}
+//ANIMAL
+fn animal(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const ANIMALS: &[&str] = &[
+        "Bear","Beetle","Carp","Cat","Cormorant","Cow","Deer","Dog","Fox",
+        "Frog","Goat","Hart","Hawk","Heron","Horse","Hound","Lion","Magpie",
+        "Owl","Panther","Peacock","Phoenix", "Rabbit","Ram","Rat","Raven","Salamander",
+        "Scorpion","Rat","Rabbit","Snake","Spider","Squirrel","Stag","Tiger",
+        "Toad","Tortoise","Turtle","Vulture","Wolf","Beetle","Locust"
+    ];
+    ANIMALS[rng.gen_range(0..ANIMALS.len())]
+}
+//DIVINE CONCEPT are more abstract stuff that doesn't go well with "the" in front of it.
+fn concept(rng: &mut impl Rng) -> &'static str {
+    #[rustfmt::skip]
+    const CONCEPTS: &[&str] = &[
+        "Love","Knowledge","Wisdom","Truth","Justice","Mercy","Protection","Healing","Strength","Courage",
+        "Fortune","Prosperity","Storms","Fire","Water","Earth","Air","Dreams","Music","Poetry","Dance",
+        "Ancestors","Transcendence","Anguish","Blight","Confessions","Connections","Courage","Decay",
+        "Lore","Silence","Triumph","Wisdom","Mending","Healing","Judgement","Forgiveness","Justice","Textiles", 
+    ];
+    CONCEPTS[rng.gen_range(0..CONCEPTS.len())]
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn name_test() {
+        let mut rng = SmallRng::seed_from_u64(0);
+
+        assert_eq!(
+            [
+                "Shrine of The Wolf",
+                "Shrine of Courage",
+                "Shrine of Foul Decay",
+                "The Singing Cave",
+                "The Fading Basin",
+                "The Exalted Gate",
+                "The Creeping Shrine",
+                "The Alabaster Shrine",
+                "Pillar of the Five Deer",
+                "The Pale Pagoda",
+                "The Immortal Shrine",
+                "Place where the Tigers Sing",
+                "Tree of Shining Textiles",
+                "The Hunting Shrine",
+                "The Creeping Shrine",
+                "Shrine of the Thirty-Six Snakes",
+                "Shrine of The Iron Deer",
+                "The Spirit Altar",
+                "Shrine of Forgiveness",
+                "The Dark Shrine"
```

Comment:
```suggestion
                "Shrine of the Wolf",
                "Shrine of Courage",
                "Shrine of Foul Decay",
                "The Singing Cave",
                "The Fading Basin",
                "The Exalted Gate",
                "The Creeping Shrine",
                "The Alabaster Shrine",
                "Pillar of the Five Deer",
                "The Pale Pagoda",
                "The Immortal Shrine",
                "Place Where the Tigers Sing",
                "Tree of Shining Textiles",
                "The Hunting Shrine",
                "The Creeping Shrine",
                "Shrine of the Thirty-Six Snakes",
                "Shrine of the Iron Deer",
                "The Spirit Altar",
                "Shrine of Forgiveness",
                "The Dark Shrine"
```

The capitalization seems a bit random here.